Lib/test/test_xpickle.py

```diff
@@ -0,0 +1,260 @@
+    Otherwise (and for all other platforms), it will attempt to check for
+    python<py_version[0]>.<py_version[1]>.
+
+    Eg. given a *py_version* of (3, 7), the function will attempt to try
+    'py -3.7' (for Windows) first, then 'python3.7', and return
+    ['py', '-3.7'] (on Windows) or ['python3.7'] on other platforms.
+
+    Args:
+        py_version: a 2-tuple of the major, minor version. Eg. python 3.7 would
+                    be (3, 7)
+    Returns:
+        List/Tuple containing the Python binary name and its required arguments,
+        or None if no valid binary names found.
+    """
+    python_str = ".".join(map(str, py_version))
+    targets = [('py', f'-{python_str}'), (f'python{python_str}',)]
+    if py_version not in py_executable_map:
+        with open(os.devnull, 'w') as devnull:
+            for target in targets[0 if is_windows else 1:]:
+                worker = subprocess.Popen([*target, '-c','import test.support'],
+                                          stdout=devnull,
+                                          stderr=devnull,
+                                          shell=is_windows)
+                worker.communicate()
+                if worker.returncode == 0:
+                    py_executable_map[py_version] = target
+
+    return py_executable_map.get(py_version, None)
+
+
+
+class AbstractCompatTests(PyPicklerTests):
+    py_version = None
+    _OLD_HIGHEST_PROTOCOL = pickle.HIGHEST_PROTOCOL
+
+    def setUp(self):
+        self.assertTrue(self.py_version)
+        if not have_python_version(self.py_version):
+            py_version_str = ".".join(map(str, self.py_version))
+            self.skipTest(f'Python {py_version_str} not available')
+
+        # Override the default pickle protocol to match what xpickle worker
+        # will be running.
+        highest_protocol = highest_proto_for_py_version(self.py_version)
+        pickletester.protocols = range(highest_protocol + 1)
+        pickle.HIGHEST_PROTOCOL = highest_protocol
+
+    def tearDown(self):
+        # Set the highest protocol back to the default.
+        pickletester.protocols = range(pickle.HIGHEST_PROTOCOL + 1)
+        pickle.HIGHEST_PROTOCOL = self._OLD_HIGHEST_PROTOCOL
+
+    def send_to_worker(self, python, obj, proto, **kwargs):
+        """Bounce a pickled object through another version of Python.
+        This will pickle the object, send it to a child process where it will
+        be unpickled, then repickled and sent back to the parent process.
+        Args:
+            python: list containing the python binary to start and its arguments
+            obj: object to pickle.
+            proto: pickle protocol number to use.
+            kwargs: other keyword arguments to pass into pickle.dumps()
+        Returns:
+            The pickled data received from the child process.
+        """
+        target = pathlib.Path(__file__).parent / 'xpickle_worker.py'
+        data = super().dumps((proto, obj), proto, **kwargs)
```

Comment:
I think it's clearer/cleaner if super().dumps is called from self.dumps directly.  What do you think about moving this call to self.dumps and passing the data as arg to send_to_worker?

Reply:
While I don't have any strong opinion on this, I do see what you mean. send_to_worker should probably only be responsible for sending data to a worker and receiving it,  super().dumps seems to be a side thing rather than the role of the worker.